### ISSUE TITLE
offer baked badge download after claim

### ISSUE THREAD

**Issue by stenington** [mozilla/openbadges-badges#37]
After #36, this will be offered from the "store" panel on the claim page.

I don't think `openbadges-badges` should actually do the baking... either openbadger should, or we should hit the bakery/backpack baking service.

**Comment by stenington** [comment]
We could also potentially use https://github.com/toolness/openbadges-instabaker.js although I'm a little concerned about what happens with non-modern browsers.

**Comment by cmcavoy** [comment]
We could switch it off for old browsers...

On Thu, Nov 21, 2013 at 4:35 PM, Mike Larsson notifications@github.comwrote:

> We could also potentially use
> https://github.com/toolness/openbadges-instabaker.js although I'm a
> little concerned about what happens with non-modern browsers.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/mozilla/openbadges-badges/issues/37#issuecomment-29031619
> .

## 

@chmcavoy http://twitter.com/chmcavoy
http://lonelylion.com

**Comment by stenington** [comment]
Yeah, so I think the way it's written is that it doesn't try to bake until you mouse over the image. It might be odd to offer a "download" button that changes to "nevermind" when they mouse over and baking fails, or whatever.

I guess maybe we could make the mouseover behavior configurable and just attempt to bake immediately, bailing out and showing nothing if it fails.

**Comment by iamjessklein** [comment]
Is the reason behind your thinking processing speed or something else?

**Comment by stenington** [comment]
I _think_ the logic behind baking on mouseover was not to overly tax the client if a page loads 500 badges by baking all of them immediately, even if some are hidden or all the user wants to do is go to the next page or whatever, but bake "on demand" as it were.

However in this case there will just be one badge on the page, so baking immediately seems fine.

**Comment by threeqube** [comment]
> However in this case there will just be one badge on the page, so baking immediately seems fine.

Agreed.

**Comment by xmatthewx** [comment]
@stenington Is it possible to check whether someone has a backpack before loading this screen? If so, we could frame downloading and pushing to backpack differently.
